src/js/vue/mainVue.js

```diff
@@ -85,6 +88,56 @@ MainVue.showSearchModal = function (options) {
     app.modalOptions = options || {};
 };
 
+/**
+ * show message box (generic method for all message types)
+ * @param options.header header text (will be translated)
+ * @param options.message message text
+ * @param options.items array of items to show as list
+ * @param options.autoCloseDuration duration in ms before auto-close (default: 0 = no auto-close)
+ * @param options.type modal type: 'success', 'question', 'warning', 'info' (default: 'success')
+ * @param options.buttonPreset button preset: 'ok', 'yesno', 'okcancel'
+ * @param options.buttons custom buttons array
+ * @param options.showCloseButton show X close button (default: true)
+ * @param options.onClose callback function called when modal closes
+ * @returns Promise that resolves with button value
+ */
+MainVue.showMessageBox = function (options) {
+    if (!app) {
+        return Promise.resolve(false);
+    }
+    app.showModal = modalTypes.MODAL_SUCCESS;
+    return new Promise((resolve) => {
+        app.$nextTick(() => {
+            if (app.$refs.messageBox) {
+                app.$refs.messageBox.show(options || {}).then(resolve);
+            } else {
+                resolve(false);
+            }
+        });
+    });
+};
+
+/**
+ * show confirmation dialog (replaces native confirm())
+ * @param message confirmation message
+ * @param options.header header text
+ * @param options.buttonPreset button preset: 'yesno' (default), 'okcancel', 'ok'
+ * @returns Promise that resolves to true if confirmed, false if cancelled
+ */
+MainVue.showConfirmBox = function (message, options = {}) {
+    return MainVue.showMessageBox({
+        type: 'question',
+        header: options.header,
+        message: message,
+        buttonPreset: options.buttonPreset || 'yesno',
+        showCloseButton: options.showCloseButton !== undefined ? options.showCloseButton : true
+    });
+};
+
+// Backward compatibility aliases
+MainVue.showSuccessModal = MainVue.showMessageBox;
```

Comment:
please remove - we don't need backward compatibilty for something which was never in production.